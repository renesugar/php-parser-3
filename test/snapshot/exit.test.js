const parser = require('../main');

describe("exit", function() {
  it("simple", function() {
    expect(parser.parseEval('exit();')).toMatchSnapshot();
  });
  it("argument", function() {
    expect(parser.parseEval('exit($var);')).toMatchSnapshot();
  });
  it("die", function() {
    expect(parser.parseEval('die();')).toMatchSnapshot();
  });
});
