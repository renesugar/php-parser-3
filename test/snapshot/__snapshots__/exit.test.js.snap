// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exit argument 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": Exit {
        "kind": "exit",
        "status": Variable {
          "byref": false,
          "curly": false,
          "kind": "variable",
          "name": "var",
        },
        "useDie": false,
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`exit die 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": Exit {
        "kind": "exit",
        "status": null,
        "useDie": true,
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`exit simple 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": Exit {
        "kind": "exit",
        "status": null,
        "useDie": false,
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;
